components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AuthenticationDecisionEvaluator2.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authentication.framework;
+
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
+
+/**
+ * Definition for dynamic Authentication Decision evaluation.
+ * Custom functions can be added supporting this contract to the dynamic sequence handler.
+ *
+ * @see org.wso2.carbon.identity.application.authentication.framework.handler.sequence.impl.GraphBasedSequenceHandler
+ */
+@FunctionalInterface
+public interface AuthenticationDecisionEvaluator2 {
```

Comment:
once deprecated AuthenticationDecisionEvaluator is removed this will be AuthenticationDecisionEvaluator